src/sdc11073/pysoap/soapclientpool.py

```diff
@@ -1,33 +1,39 @@
-from typing import Protocol, Callable, Any
-from .. import loghelper
+from __future__ import annotations
 
+from typing import TYPE_CHECKING, Any, Callable, Protocol
+
+from sdc11073 import loghelper
+
+if TYPE_CHECKING:
+    from .soapclient import SoapClientProtocol
 
 _SoapClientFactory = Callable[[str, list[str]], Any]
 
 
 class UserRef(Protocol):
-    def on_unreachable(self):
-        ...
+    """Reference counter that notifies reference holders when net location is unreachable."""
 
+    def on_unreachable(self):
+        """Notify net location is unreachable."""
 
 
 class _SoapClientEntry:
```

Comment:
make dataclass

Reply:
no, member "callbacks"is not the same as argument "unreachable_callback".